pytest_html/resources/main.js

```diff
@@ -2,6 +2,12 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+//created to increase testability
+var  DOCUMENT = document;
+
+function setDocument(elem) {
+  DOCUMENT = elem;
+}
```

Comment:
This is also a pretty good resource with some suggestions: https://qunitjs.com/intro/#testing-the-dom-manipulation